bundle/edu.gemini.shared.gui/src/main/scala/edu/gemini/shared/gui/textComponent/NumberField.scala

```diff
@@ -27,7 +27,7 @@ class NumberField(d: Option[Double]) extends FormattedTextField(NumberField.df)
   private val pink = new Color(255, 224, 224)
   private val white = background // N.B. this means we can't change the background to anything else
 
-  var valid = true
+  @volatile var valid = true
```

Comment:
Could you explain a bit what prompted the `volatile` annotation?  On the surface it is confusing because Swing is single threaded anyway.  If it is being accessed from another thread perhaps it should be shunted into the event loop with `Swing.onEDT` or something?